### ISSUE TITLE
[java-generator] "package" is not handled as keyword

### ISSUE THREAD

**Issue by Obirah** [fabric8io/kubernetes-client#5380]
### Describe the bug

When using `io.fabric8.java-generator` to genreate Java code from a CRD which contains an object that is named package `package` and has objects underneath it, the generator simply uses the name `package` as package name even though it is a Java keyword which results in broken code.

### Fabric8 Kubernetes Client version

6.8.0

### Steps to reproduce

Create a CRD which contains an object named `package` anywhere in the Spec or Status.

Example for such an object: 
```
package:
  description: Package to be used.
  properties:
    appPath:
      description: AppPath of the app relative to the package's root.
      type: string
    name:
      description: Name of the package.
      type: string
    repository:
      description: Repository to be used to obtain the package.
      properties:
        secretName:
          description: SecretName to be used to access the registry
            (must be in the same namespace).
          type: string
        type:
          description: Type of the Repository.
          enum:
          - git
          - npm
          type: string
        uri:
          description: URI for the registry.
          pattern: ^\w+:(\/?\/?)[^\s]+$
          type: string
      required:
      - type
      - uri
      type: object
    version:
      description: Version of the package.
      type: string
  required:
  - repository
  - version
  type: object
```

Generate code for your CRD and you will see that the package name for the nested object `repository` will look something like this: `com.acme.mycr.package` which will not be compilable.

### Expected behavior

Expected behavior would be to have compilable Java code even when an object is called `package`. 

In my mind, one possible solution would be to prepend the name of the parent package to the package name if a package is called `package`, e.g. instead of `com.acme.mycr.package` the generator produces `com.acme.mycr.mycrpackage`.

Another solution would be to throw an error if a keyword occurs and make the users specify `keywordOverrides` similar to the `packageOverrides`.

### Runtime

Kubernetes (vanilla)

### Kubernetes API Server version

1.25.3@latest

### Environment

Linux

**Comment by andreaTP** [comment]
I'm actually surprised as we are already sanitizing strings and excluding keywords, but I can reproduce the issue with a CRD like:

```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: dummies.example.com
spec:
  group: example.com
  versions:
    - name: v1
      served: true
      storage: true
      schema:
        openAPIV3Schema:
          type: object
          properties:
            spec:
              type: object
              properties:
                package:
                  type: object
                  properties:
                    foo:
                      type: object
                      properties:
                        bar:
                          type: string
  scope: Namespaced
  names:
    plural: dummies
    singular: dummy
    kind: Dummy
```

**Comment by andreaTP** [comment]
@Obirah thanks again for reporting, I submitted a fix in #5381 let me know if you have any comment.

**Comment by andreaTP** [comment]
> I wouldn't have implemented a post-processor that fixes this in Gradle. :D

Next time I'll assign the issue to you 😜 😄 !
